tls/s2n_signature_algorithms.c

```diff
@@ -182,21 +185,42 @@ int s2n_choose_sig_scheme_from_peer_preference_list(struct s2n_connection *conn,
     return 0;
 }
 
-int s2n_send_supported_signature_algorithms(struct s2n_stuffer *out)
+int s2n_send_supported_signature_schemes(struct s2n_connection *conn, struct s2n_stuffer *out)
 {
-    /* The array of hashes and signature algorithms we support */
-    uint16_t num_signature_schemes = s2n_supported_sig_scheme_pref_list_len;
-    uint16_t signature_schemes_size = num_signature_schemes * TLS_SIGNATURE_SCHEME_LEN;
+    const struct s2n_signature_preferences *signature_preferences = conn->config->signature_preferences;
+    notnull_check(signature_preferences);
 
-    GUARD(s2n_stuffer_write_uint16(out, signature_schemes_size));
+    GUARD(s2n_stuffer_write_uint16(out, s2n_supported_signature_schemes_size(conn)));
 
-    for (int i =  0; i < num_signature_schemes; i++) {
-        GUARD(s2n_stuffer_write_uint16(out, s2n_supported_sig_scheme_pref_list[i]->iana_value));
+    for (int i =  0; i < signature_preferences->count; i++) {
+        const struct s2n_signature_scheme *const scheme = signature_preferences->signature_schemes[i];
+        if (0 == s2n_signature_scheme_valid_to_offer(conn, scheme)) {
+            GUARD(s2n_stuffer_write_uint16(out, scheme->iana_value));
+        }
     }
 
     return 0;
 }
 
+int s2n_supported_signature_schemes_size(struct s2n_connection *conn)
+{
+    return s2n_supported_signature_schemes_length(conn) * TLS_SIGNATURE_SCHEME_LEN;
+}
+
+int s2n_supported_signature_schemes_length(struct s2n_connection *conn)
```

Comment:
I think `s2n_supported_signature_schemes_count` would be more descriptive because in the above function you can write:

```c
return s2n_supported_signature_schemes_count(conn) * TLS_SIGNATURE_SCHEME_LEN;
```

If s2n is pretty standardized on _length though, better to be consistent.

Reply:
That's a much better name, will do.